### PR TITLE
Added InfluxDB source and destination

### PR DESCRIPTION
**Pull request by Jayclifford345**
This pull request contains the community contributions for InfluxDB 3.0's source and destination.

These are early beta and pretty crude but will be iterated over overtime as 3.0 develops.

Source: This plugin queries a specific measurement within InfluxDB based on a user-defined interval. Converts the result to a data frame and writes the result to a stream

Destination: Writes a data frame back to InfluxDB. Provides environment variables for defining tags and measurement name or defaults to stream name.

### CODE REVIEW

File: python/sources/InfluxDB/main.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+
+
+client = InfluxDBClient3.InfluxDBClient3(token=os.environ["INFLUXDB_TOKEN"],
+                         host=os.environ["INFLUXDB_HOST"],
+                         org=os.environ["INFLUXDB_ORG"],
+                         database=os.environ["INFLUXDB_DATABASE"])
+
+
+measurement_name = os.environ.get('INFLUXDB_MEASUREMENT_NAME', 'MEASUREMENT')
```
Comment (stereosky) [inline python/sources/InfluxDB/main.py:68]:
```suggestion
measurement_name = os.environ.get("INFLUXDB_MEASUREMENT_NAME", os.environ["output"])
```

File: python/sources/InfluxDB/main.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+interval = os.environ.get('task_interval', '5m')
```
Comment (stereosky) [inline python/sources/InfluxDB/main.py:69]:
```suggestion
interval = os.environ.get("task_interval", "5m")
```

File: python/destinations/InfluxDB/library.json [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      "InputType": "FreeText",
+      "Description": "Columns to be used as tags when writing data to InfluxDB.",
+      "DefaultValue": "['tag1', 'tag2']",
+      "Required": false
+    },
+    {
+      "Name": "INFLUXDB_MEASUREMENT_NAME",
+      "Type": "Placeholder",
+      "InputType": "FreeText",
+      "Description": "The InfluxDB measurment to write data to. If not specified, the name of the input topic will be used",
```
Comment (stereosky) [inline python/destinations/InfluxDB/library.json:70]:
```suggestion
      "Description": "The InfluxDB measurement to write data to. If not specified, the name of the input topic will be used",
```

File: python/sources/InfluxDB/library.json [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+      "InputType": "FreeText",
+      "Description": "Organization name in InfluxDB.",
+      "DefaultValue": "<ORG>",
+      "Required": false
+    },
+    {
+      "Name": "INFLUXDB_DATABASE",
+      "Type": "Placeholder",
+      "InputType": "FreeText",
+      "Description": "Database name in InfluxDB where data should be stored.",
```
Comment (stereosky) [inline python/sources/InfluxDB/library.json:62]:
```suggestion
      "Description": "Database name in InfluxDB where data is stored.",
```

File: python/sources/InfluxDB/library.json [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+      "DefaultValue": "<DATABASE>",
+      "Required": true
+    },
+    {
+      "Name": "INFLUXDB_MEASUREMENT_NAME",
+      "Type": "Placeholder",
+      "InputType": "FreeText",
+      "Description": "The InfluxDB measurment to write data to. If not specified, the name of the input topic will be used",
```
Comment (stereosky) [inline python/sources/InfluxDB/library.json:70]:
```suggestion
      "Description": "The InfluxDB measurement to read data from. If not specified, the name of the output topic will be used",
```

File: python/sources/InfluxDB/requirements.txt [hunk 6]
```diff
@@ -0,0 +1,3 @@
+quixstreams==0.5.2
+influxdb3-python
+pandas
```
Comment (stereosky) [inline python/sources/InfluxDB/requirements.txt:3]: Any chance you could provide explicit version numbers for these libraries?

File: python/sources/InfluxDB/library.json [hunk 7]
```diff
@@ -0,0 +1,75 @@
+{
+  "libraryItemId": "",
+  "name": "InfluxDB-3.0",
+  "language": "Python",
+  "tags": {
+    "Complexity": ["Easy"],
+    "Technology": ["QuixStreams"],
+    "Pipeline Stage": ["Destination"],
```
Comment (stereosky) [inline python/sources/InfluxDB/library.json:8]:
```suggestion
    "Pipeline Stage": ["Source"],
```

File: python/sources/InfluxDB/library.json [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+  "DefaultFile": "main.py",
+  "EntryPoint": "build/dockerfile",
+  "RunEntryPoint": "main.py",
+  "IconFile": "icon.png",
+  "Variables": [
+    {
+      "Name" : "output",
+      "Type" : "EnvironmentVariable",
+      "InputType" : "OutputTopic",
+      "Description" : "This is the Quix Topic that will receive the stream",
```
Comment (stereosky) [inline python/sources/InfluxDB/library.json:22]:
```suggestion
      "Description" : "This is the Quix topic that will receive the stream",
```

File: python/destinations/InfluxDB/library.json [hunk 9]
```diff
@@ -0,0 +1,75 @@
+{
+  "libraryItemId": "",
```
Comment (SteveRosam) [inline python/destinations/InfluxDB/library.json:2]: libraryItemId needs to be a valid, unique GUID.
You can generate one using https://guidgenerator.com/online-guid-generator.aspx
Same issue with the source connector

**Review by SteveRosam — CHANGES_REQUESTED** [review]

File: python/destinations/InfluxDB/library.json [hunk 10]
```diff
@@ -0,0 +1,75 @@
+{
+  "libraryItemId": "bcc847c2-f8a5-47c5-9291-45e95ce9777a",
+  "name": "InfluxDB-3.0",
```
Comment (stereosky) [inline python/destinations/InfluxDB/library.json:3]:
```suggestion
  "name": "InfluxDB 3.0",
```

File: python/sources/InfluxDB/library.json [hunk 11]
```diff
@@ -0,0 +1,75 @@
+{
+  "libraryItemId": "3f9f4c8d-9590-456b-a7c1-89daa1155a6c",
+  "name": "InfluxDB-3.0",
```
Comment (stereosky) [inline python/sources/InfluxDB/library.json:3]:
```suggestion
  "name": "InfluxDB 3.0",
```

File: python/destinations/InfluxDB/library.json [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+      "Name": "input",
+      "Type": "EnvironmentVariable",
+      "InputType": "InputTopic",
+      "Description": "This is the input topic",
+      "DefaultValue": "detection-result",
+      "Required": true
+    },
+    {
+      "Name": "INFLUXDB_HOST",
+      "Type": "Placeholder",
```
Comment (stereosky) [inline python/destinations/InfluxDB/library.json:28]:
```suggestion
      "Type": "EnvironmentVariable",
```

File: python/destinations/InfluxDB/library.json [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+      "InputType": "FreeText",
+      "Description": "Host address for the InfluxDB instance.",
+      "DefaultValue": "eu-central-1-1.aws.cloud2.influxdata.com",
+      "Required": true
+    },
+    {
+      "Name": "INFLUXDB_TOKEN",
+      "Type": "Placeholder",
```
Comment (stereosky) [inline python/destinations/InfluxDB/library.json:36]:
```suggestion
      "Type": "EnvironmentVariable",
```

File: python/destinations/InfluxDB/library.json [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+      "InputType": "FreeText",
+      "Description": "Authentication token to access InfluxDB.",
+      "DefaultValue": "<TOKEN>",
+      "Required": true
+    },
+    {
+      "Name": "INFLUXDB_ORG",
+      "Type": "Placeholder",
```
Comment (stereosky) [inline python/destinations/InfluxDB/library.json:44]:
```suggestion
      "Type": "EnvironmentVariable",
```

File: python/destinations/InfluxDB/library.json [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+      "InputType": "FreeText",
+      "Description": "Organization name in InfluxDB.",
+      "DefaultValue": "<ORG>",
+      "Required": false
+    },
+    {
+      "Name": "INFLUXDB_DATABASE",
+      "Type": "Placeholder",
```
Comment (stereosky) [inline python/destinations/InfluxDB/library.json:52]:
```suggestion
      "Type": "EnvironmentVariable",
```

File: python/destinations/InfluxDB/library.json [hunk 16]
```diff
@@ -0,0 +1,75 @@
...
+      "InputType": "FreeText",
+      "Description": "Database name in InfluxDB where data should be stored.",
+      "DefaultValue": "<DATABASE>",
+      "Required": true
+    },
+    {
+      "Name": "INFLUXDB_TAG_COLUMNS",
+      "Type": "Placeholder",
```
Comment (stereosky) [inline python/destinations/InfluxDB/library.json:60]:
```suggestion
      "Type": "EnvironmentVariable",
```

File: python/destinations/InfluxDB/library.json [hunk 17]
```diff
@@ -0,0 +1,75 @@
...
+      "InputType": "FreeText",
+      "Description": "Columns to be used as tags when writing data to InfluxDB.",
+      "DefaultValue": "['tag1', 'tag2']",
+      "Required": false
+    },
+    {
+      "Name": "INFLUXDB_MEASUREMENT_NAME",
+      "Type": "Placeholder",
```
Comment (stereosky) [inline python/destinations/InfluxDB/library.json:68]:
```suggestion
      "Type": "EnvironmentVariable",
```

File: python/sources/InfluxDB/library.json [hunk 18]
```diff
@@ -0,0 +1,75 @@
...
+      "Name" : "output",
+      "Type" : "EnvironmentVariable",
+      "InputType" : "OutputTopic",
+      "Description" : "This is the Quix topic that will receive the stream",
+      "DefaultValue" : "influxdb",
+      "Required": true
+  },
+    {
+      "Name": "task_interval",
+      "Type": "Placeholder",
```
Comment (stereosky) [inline python/sources/InfluxDB/library.json:28]:
```suggestion
      "Type": "EnvironmentVariable",
```

File: python/sources/InfluxDB/library.json [hunk 19]
```diff
@@ -0,0 +1,75 @@
...
+      "InputType": "FreeText",
+      "Description": "Interval to run query. Must be within the InfluxDB notation; 1s, 1m, 1h, 1d, 1w, 1mo, 1y",
+      "DefaultValue": "5m",
+      "Required": true
+    },
+    {
+      "Name": "INFLUXDB_HOST",
+      "Type": "Placeholder",
```
Comment (stereosky) [inline python/sources/InfluxDB/library.json:36]:
```suggestion
      "Type": "EnvironmentVariable",
```

File: python/sources/InfluxDB/library.json [hunk 20]
```diff
@@ -0,0 +1,75 @@
...
+      "InputType": "FreeText",
+      "Description": "Host address for the InfluxDB instance.",
+      "DefaultValue": "eu-central-1-1.aws.cloud2.influxdata.com",
+      "Required": true
+    },
+    {
+      "Name": "INFLUXDB_TOKEN",
+      "Type": "Placeholder",
```
Comment (stereosky) [inline python/sources/InfluxDB/library.json:44]:
```suggestion
      "Type": "EnvironmentVariable",
```

File: python/sources/InfluxDB/library.json [hunk 21]
```diff
@@ -0,0 +1,75 @@
...
+      "InputType": "FreeText",
+      "Description": "Authentication token to access InfluxDB.",
+      "DefaultValue": "<TOKEN>",
+      "Required": true
+    },
+    {
+      "Name": "INFLUXDB_ORG",
+      "Type": "Placeholder",
```
Comment (stereosky) [inline python/sources/InfluxDB/library.json:52]:
```suggestion
      "Type": "EnvironmentVariable",
```

File: python/sources/InfluxDB/library.json [hunk 22]
```diff
@@ -0,0 +1,75 @@
...
+      "InputType": "FreeText",
+      "Description": "Organization name in InfluxDB.",
+      "DefaultValue": "<ORG>",
+      "Required": false
+    },
+    {
+      "Name": "INFLUXDB_DATABASE",
+      "Type": "Placeholder",
```
Comment (stereosky) [inline python/sources/InfluxDB/library.json:60]:
```suggestion
      "Type": "EnvironmentVariable",
```

File: python/sources/InfluxDB/library.json [hunk 23]
```diff
@@ -0,0 +1,75 @@
...
+      "InputType": "FreeText",
+      "Description": "Database name in InfluxDB where data is stored.",
+      "DefaultValue": "<DATABASE>",
+      "Required": true
+    },
+    {
+      "Name": "INFLUXDB_MEASUREMENT_NAME",
+      "Type": "Placeholder",
```
Comment (stereosky) [inline python/sources/InfluxDB/library.json:68]:
```suggestion
      "Type": "EnvironmentVariable",
```

**Review by stereosky — CHANGES_REQUESTED** [review]
Remove dash from name and change Type value from `Placeholder` to `EnvironmentVariable` to make the widget display correctly in the UI

File: python/destinations/InfluxDB/README.md [hunk 24]
```diff
@@ -0,0 +1,38 @@
+
+
+# InfluxDB 3.0
+
+Write data to InfluxDB 3.0 with [this project](https://github.com/quixio/quix-samples/tree/main/python/destinations/InfluxDB).
+
+## How to run
+
+Create a [Quix](https://portal.platform.quix.ai/self-sign-up?xlink=github) account or log-in and visit the Samples to use this project.
```
Comment (stereosky) [inline python/destinations/InfluxDB/README.md:9]:
```suggestion
Create a [Quix](https://portal.platform.quix.ai/self-sign-up?xlink=github) account or log in and visit the Code Samples to use this project.
```

File: python/sources/InfluxDB/README.md [hunk 25]
```diff
@@ -0,0 +1,36 @@
+# InfluxDB 3.0
+
+Read data from InfluxDB 3.0 with [this project](https://github.com/quixio/quix-samples/tree/main/python/sources/InfluxDB).
+
+## How to run
+
+Create a [Quix](https://portal.platform.quix.ai/self-sign-up?xlink=github) account or log-in and visit the Samples to use this project.
```
Comment (stereosky) [inline python/sources/InfluxDB/README.md:7]:
```suggestion
Create a [Quix](https://portal.platform.quix.ai/self-sign-up?xlink=github) account or log in and visit the Code Samples to use this project.
```

**Comment by Jayclifford345** [comment]
@stereosky, @SteveRosam  sorry for the late approval of your changes. I was on holiday, let me know if changes look okay.

File: python/sources/InfluxDB/library.json [hunk 26]
```diff
@@ -0,0 +1,75 @@
...
+      "DefaultValue": "<TOKEN>",
+      "Required": true
+    },
+    {
+      "Name": "INFLUXDB_ORG",
+      "Type": "EnvironmentVariable",
+      "InputType": "FreeText",
+      "Description": "Organization name in InfluxDB.",
+      "DefaultValue": "<ORG>",
+      "Required": false
```
Comment (srosam) [inline python/sources/InfluxDB/library.json:56]: I feel like this should be mandatory. But if it's really optional then pls ignore this comment.

File: python/destinations/InfluxDB/library.json [hunk 27]
```diff
@@ -0,0 +1,75 @@
...
+      "Description": "Columns to be used as tags when writing data to InfluxDB.",
+      "DefaultValue": "['tag1', 'tag2']",
+      "Required": false
+    },
+    {
+      "Name": "INFLUXDB_MEASUREMENT_NAME",
+      "Type": "EnvironmentVariable",
+      "InputType": "FreeText",
+      "Description": "The InfluxDB measurement to write data to. If not specified, the name of the input topic will be used",
+      "DefaultValue": "<INSERT MEASUREMENT>",
```
[the remark this answers is not in the document]
Comment (Jayclifford345) [reply python/destinations/InfluxDB/library.json]: Okay, I have removed the default placeholders apart from the tag columns so users can understand the expected structure.

File: python/sources/InfluxDB/library.json (hunk 26 above)
Comment (Jayclifford345) [reply python/sources/InfluxDB/library.json:56]: I might remove this ORG environment variable altogether as with 3.0 you do not need it to query only to write.

**Comment by Jayclifford345** [comment]
Hi @SteveRosam @stereosky, that should be changes reviewed and implemented

**Comment by srosam** [comment]
> Hi @SteveRosam @stereosky, that should be changes reviewed and implemented

Awesome, thanks for making those changes. Looks great!

**Review by srosam — APPROVED** [review]

**Pull request merged**